index.html

```diff
@@ -961,13 +961,26 @@ <h2><dfn>Coalesced events</dfn></h2>
                 of the pointer movement.</figcaption>
             </figure>
 
-            <p>A <a>PointerEvent</a> has an associated <dfn>coalesced event list</dfn> (a list of
+            <p>A <a>PointerEvent</a> has an associated <dfn>coalesced events list</dfn> (a list of
             zero or more <code>PointerEvent</code>s). If this event is a <code>pointermove</code>
             or <code>pointerrawupdate</code> event, the list is a sequence of all <code>PointerEvent</code>s
             that were coalesced into this event; otherwise it is an empty list.</p>
 
-            <p>The events in the coalesced event list will have increasing
-            <a href="https://dom.spec.whatwg.org/#dom-event-timestamp"><code>timeStamps</code></a>
+            <p>When the <code>getCoalescedEvents()</code> method is called, user agents MUST apply the following
+               steps to each event in the coalesced events list, before returning the list:</p>
+            <ul>
+                <li>set <code>pointerId</code>, <code>pointerType</code>, <code>isPrimary</code>,
+                    <a href="https://dom.spec.whatwg.org/#dom-event-istrusted"><code>isTrusted</code></a>,
+                    and <a href="https://dom.spec.whatwg.org/#event-target"><code>target</code></a> to match the
+                    dispatched "parent" pointer event that the <code>getCoalescedEvents()</code> method was called on</li>
```

Comment:
In the points which are coalesced I think the only thing that could vary is the target (as the pointer moves over different targets). We should never coalesce events from other pointers into a different pointer's event. As such, I think we only need to call out target here.

Reply:
but then it would need to say somewhere (else?) that the coalesced / predicted events have the same id/type/etc as their "parent" event that the method was called on (otherwise authors don't know what those values would be set to?)

Reply:
adding here that, from what @smaug---- was saying, in the Gecko implementation the non-coalesced "raw" events are not stored as full-fledged pointer events internally, but some more general kind of event. only when `getCoalescedEvents` is called are they turned into pointer events for the list that is then returned. so probably from that point of view, these steps make more sense. if an engine instead already stores them up as if they were real pointer events, and then only needs to ensure that they're retargeted to the "parent" pointer event where the method is actually called, then yes only retargetting step here would be needed. but then we may need to split out somehting about all the other properties? a separate list that explains what authors can expect `pointerId` etc to be (that they'll also match the "parent" event)

Reply:
> but then it would need to say somewhere (else?) that the coalesced / predicted events have the same id/type/etc as their "parent" event that the method was called on (otherwise authors don't know what those values would be set to?)

This seems reasonable to add. We already state that "The events in the coalesced event list will have increasing timeStamps," we could add to this that those other properties will be the same as the PointerEvent (the root one).

> adding here that, from what @smaug---- was saying, in the Gecko implementation the non-coalesced "raw" events are not stored as full-fledged pointer events internally, but some more general kind of event. only when `getCoalescedEvents` is called are they turned into pointer events for the list that is then returned. so probably from that point of view, these steps make more sense. if an engine instead already stores them up as if they were real pointer events, and then only needs to ensure that they're retargeted to the "parent" pointer event where the method is actually called, then yes only retargetting step here would be needed. but then we may need to split out somehting about all the other properties? a separate list that explains what authors can expect `pointerId` etc to be (that they'll also match the "parent" event)

The spec already seems to imply that the PointerEvent keeps a list of full PointerEvents in the coalesced events list, "A PointerEvent has an associated coalesced event list (a list of zero or more PointerEvents)." Implementations can of course optimize and build this on the fly as long as the side effects (i.e. calls to getCoalescedEvents) return a list that is consistent with this assumption. That does raise the question of whether you should get the same list of events with multiple calls - I'd say the spec sort of implies that you should right now, but perhaps it shouldn't.